File: python/euclideanAlgorithm.py
def gcd(a, b):
  while b != 0:
    a, b = b, a % b
  return a 

# example usage
num1 = 48
num2 = 18
result = gcd(num1, num2)

print("gcd of", num1, "and", num2, "is", result)

File: java/EuclideanAlgorithm.java
// used to find gcd of two ints

public class EuclideanAlgorithm
  {
    public static int gcd(int a, int b)
    {
      if(b > a)
      {
        int temp = a;
        a = b;
        b = temp;
      }
      // base case: if b is 0, a is the GCD
      if(b == 0)
      {
        return a;
      }
      return gcd(b, a % b)
        }


    public static void main(String[] args)
    {
      int num1 = 48; 
      int num2 = 18;
      int result = gcd(num1, num2);

    }
  }
